### PR TITLE
docs/plugin/framework: Update code examples for v0.3.0 breaking changes

### PR DESCRIPTION
**Pull request by bflad**
Closes hashicorp/terraform-plugin-framework#140

## Labels

- [x] inaccuracy
- [ ] clarification
- [ ] new docs
- [ ] cosmetic bug - fixing broken text or markup
- [ ] enhancement - changing the website's behavior/layout
- [ ] api - requires an update to the [changelog](https://www.terraform.io/docs/cloud/api/changelog.html)

### CODE REVIEW

File: content/source/docs/plugin/framework/accessing-values.html.md [hunk 1]
```diff
@@ -43,8 +43,9 @@ type resourceData struct {
 func (m myResource) Create(ctx context.Context,
 	req tfsdk.CreateResourceRequest, resp *tfsdk.CreateResourceResponse) {
 	var plan resourceData
-	err := req.Plan.Get(ctx, &plan)
-	if err != nil {
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
 		// TODO: handle error
```
Comment (paddycarver) [inline content/source/docs/plugin/framework/accessing-values.html.md:49]: This can probably just return, now.

**Review by paddycarver — CHANGES_REQUESTED** [review]

File: content/source/docs/plugin/framework/accessing-values.html.md (hunk 1 above)
Comment (bflad) [reply content/source/docs/plugin/framework/accessing-values.html.md:49]: Indeed! Will fill that in.

File: content/source/docs/plugin/framework/accessing-values.html.md [hunk 2]
```diff
@@ -94,8 +95,9 @@ type will always be the type produced by that attribute's `attr.Type`.
 ```go
 func (m myResource) Create(ctx context.Context,
 	req tfsdk.CreateResourceRequest, resp *tfsdk.CreateResourceResponse) {
-	attr, err := req.Config.GetAttribute(ctx, tftypes.NewAttributePath().WithAttributeName("age"))
-	if err != nil {
+	attr, diags := req.Config.GetAttribute(ctx, tftypes.NewAttributePath().WithAttributeName("age"))
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
 		// TODO: handle error
```
Comment (bflad) [inline content/source/docs/plugin/framework/accessing-values.html.md:101]:
```suggestion
		return
```

File: content/source/docs/plugin/framework/writing-state.html.md [hunk 3]
```diff
@@ -44,8 +44,9 @@ func (m myResource) Create(ctx context.Context,
 	newState.Name.Value = "J. Doe"
 
 	// persist the values to state
-	err := resp.State.Set(ctx, &newState)
-	if err != nil {
+	diags := resp.State.Set(ctx, &newState)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
 		// TODO: handle error
```
Comment (bflad) [inline content/source/docs/plugin/framework/writing-state.html.md:50]:
```suggestion
		return
```

File: content/source/docs/plugin/framework/writing-state.html.md [hunk 4]
```diff
@@ -94,8 +95,9 @@ runtime.
 func (m myResource) Create(ctx context.Context,
 	req tfsdk.CreateResourceRequest, resp *tfsdk.CreateResourceResponse) {
 	age := types.Number{Value: big.NewFloat(7)}
-	err := resp.State.SetAttribute(ctx, tftypes.NewAttributePath().WithAttributeName("age"), &age)
-	if err != nil {
+	diags := resp.State.SetAttribute(ctx, tftypes.NewAttributePath().WithAttributeName("age"), &age)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
 		// TODO: handle error
```
Comment (bflad) [inline content/source/docs/plugin/framework/writing-state.html.md:101]:
```suggestion
		return
```

File: content/source/docs/plugin/framework/writing-state.html.md [hunk 5]
```diff
@@ -108,8 +110,9 @@ func (m myResource) Create(ctx context.Context,
 	req tfsdk.CreateResourceRequest, resp *tfsdk.CreateResourceResponse) {
 	var age int64
 	age = 7
-	err := resp.State.SetAttribute(ctx, tftypes.NewAttributePath().WithAttributeName("age"), &age)
-	if err != nil {
+	diags := resp.State.SetAttribute(ctx, tftypes.NewAttributePath().WithAttributeName("age"), &age)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
 		// TODO: handle error
```
Comment (bflad) [inline content/source/docs/plugin/framework/writing-state.html.md:116]:
```suggestion
		return
```

**Review by paddycarver — APPROVED** [review]
LGTM, let's do it. Nice work! 👍 🚀

**Pull request merged**